locallib.php

```diff
@@ -513,7 +513,7 @@ function block_onlinesurvey_get_lti_content($config = null, $context = null, $co
         } else {
             $reinstructor = BLOCK_ONLINESURVEY_LTI_REGEX_INSTRUCTOR_DEFAULT;
         }
-        if (empty($matches) && !empty($reinstructor)) {
+        if (!empty($reinstructor)) {
```

Comment:
Removing the empty($matches) clause seems to make sense.
However, with the patch as it stands now, you won't be able to realize the "total surveycount = surveycount as a leaner + surveycount as a instructor." goal which you mentioned in https://github.com/evasys-community/moodle-block_onlinesurvey/issues/31#issue-1265952304.
Instead, the survey count of the instructor courses will overwrite the survey count of the student courses.

Would you want to rework line 516 and 517 so that the survey counts are added if a user is a student and an instructor?

Reply:
Yes I am working on it.